### PR TITLE
Simplify git_revparse_ext usage

### PR DESCRIPTION
**Pull request by nulltoken**

### CODE REVIEW

**Comment by nulltoken** [comment]
@jamill What's your opinion about this?

File: LibGit2Sharp/Core/DisposableTuple.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        public DisposableTuple(T1 item1, T2 item2)
+        {
+            _tuple = new Tuple<T1, T2>(item1, item2);
+        }
+
+        public T1 Item1 { get { return _tuple.Item1; } }
+        public T2 Item2 { get { return _tuple.Item2; } }
+
+        public void Dispose()
```
Comment (jamill) [inline LibGit2Sharp/Core/DisposableTuple.cs]: Should we call `GC.SuppressFinalize()` at the end of this method?

Comment (nulltoken) [inline LibGit2Sharp/Core/DisposableTuple.cs]: Oh! Indeed. Thanks for this

**Comment by jamill** [comment]
`DisposableTuple` feels a bit awkward to me - it is kindof a general construct, but is tailored to a specific use case here. What would you think about just creating a private class that encapsulates the return value of `git_revparse_ext` and has 2 specific members with friendly names?

File: LibGit2Sharp/Core/DisposableTuple.cs (hunk 1 above)
Comment (jamill) [inline LibGit2Sharp/Core/DisposableTuple.cs]: Actually, I think I was mistaken here - I don't think this is needed.

From [Object.Finalize](https://msdn.microsoft.com/en-us/library/system.object.finalize(v=vs.110).aspx) documentation:
> The Object class provides no implementation for the Finalize method, and the garbage collector does not mark types derived from Object for finalization unless they override the Finalize method.

Comment (nulltoken) [inline LibGit2Sharp/Core/DisposableTuple.cs]: Fixed

**Comment by nulltoken** [comment]
> What would you think about just creating a private class that encapsulates the return value of git_revparse_ext and has 2 specific members with friendly names?

I'll try this out and report back when it's done.

**Pull request closed without merging**